diff --git a/src/screen.c b/src/screen.c
index 3dde3b4..6c55cd6 100644
--- a/src/screen.c
+++ b/src/screen.c
@@ -109,6 +109,11 @@
 
 #include "logfile.h"	/* islogfile, logfflush */
 
+#ifdef __APPLE__
+#include <vproc.h>
+#include "vproc_priv.h"
+#endif
+
 #ifdef DEBUG
 FILE *dfp;
 #endif
@@ -1303,6 +1308,11 @@ char **av;
   freopen("/dev/null", "w", stderr);
   debug("-- screen.back debug started\n");
 
+#ifdef __APPLE__
+  if (_vprocmgr_move_subset_to_user(real_uid, "Background") != NULL)
+    errx(1, "can't migrate to background session");
+#endif
+
   /* 
    * This guarantees that the session owner is listed, even when we
    * start detached. From now on we should not refer to 'LoginName'
