From e00af144f5e18046d113304a6dacec8c70cb8e60 Mon Sep 17 00:00:00 2001
From: Ivan Torres <mexpolk@gmail.com>
Date: Fri, 4 Feb 2011 04:34:35 -0600
Subject: [PATCH] Fix for snow leopard

---
 screen.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/screen.c b/screen.c
index 3dde3b4..6c55cd6 100644
--- a/screen.c
+++ b/screen.c
@@ -109,6 +109,11 @@
 
 #include "logfile.h"	/* islogfile, logfflush */
 
+#ifdef __APPLE__
+#include <vproc.h>
+#include "vproc_priv.h"
+#endif
+
 #ifdef DEBUG
 FILE *dfp;
 #endif
@@ -1303,6 +1308,11 @@ char **av;
   freopen("/dev/null", "w", stderr);
   debug("-- screen.back debug started\n");
 
+#ifdef __APPLE__
+  if (_vprocmgr_move_subset_to_user(real_uid, "Background") != NULL)
+    errx(1, "can't migrate to background session");
+#endif
+
   /* 
    * This guarantees that the session owner is listed, even when we
    * start detached. From now on we should not refer to 'LoginName'
-- 
1.7.3.5

